src/components/Layout/ApplicationBody.jsx

```diff
@@ -77,27 +98,47 @@ const ApplicationBody = () => {
         });
     };
 
-    const updateCardData = (newCard) => {
-        setCardsList((prevState) => prevState.map(el => {
-            return el.id === newCard.id ? newCard : el;
-        }));
-    };
+    const updateCardData = useCallback((newCard) => {
+        setCardsList((prevState) => {
+            const newState = prevState.map(el => (el.id === newCard.id ? newCard : el));
+
+            const newIsSomeSelected = newState.findIndex(el => (el.isSelected === true)) !== -1;
+            if (newIsSomeSelected !== isSomeSelected) setIsSomeSelected(newIsSomeSelected);
+
+            return newState;
+        });
+    }, [isSomeSelected]);
+
+    const deleteSelected = (event) => {
+        if ((event.type === 'keydown' || event.type === 'keyup') && event.code === 'Space') event.preventDefault();
+        if (event.type === 'keyup' || (event.type === 'keydown' && event.code !== 'Space')) return;
+
+        if (!isSomeSelected) return;
```

Comment:
isSomeSelected state is redundant. You can firstly do action from 118-th line and then check length of returned array.

Reply:
isSomeSelected is used by "Delete Selected" button's styles to indicate disabled state.